### PR TITLE
Feat platform 868 order and group pins

### PR DESCRIPTION
**Pull request by GabrielPadureanu**
refs: https://github.com/withthegrid/platform/issues/868
is required by: 
https://github.com/withthegrid/platform/pull/1584
https://github.com/withthegrid/platform-client/pull/1564

### CODE REVIEW

File: src/models/analytics/analytics-tables.ts [hunk 1]
```diff
@@ -532,6 +561,8 @@ const analyticsTablesL10n: Record<TableKeys, AnalyticsTable> = {
         createdAt: 'Aangemaakt op',
         updatedAt: 'Bijgewerkt op',
         deletedAt: 'Verwijderd op',
+        pinGroupSort: 'NL Order in pin group',
```
Comment (GabrielPadureanu) [inline src/models/analytics/analytics-tables.ts:564]: Please help me with a translation (maybe with another version in English too if you have a nicer idea)

**Comment by srieding** [comment]
This would have been marginally nicer if it followed the branch naming convention from the readme: `feat-[repo-name]-[ticket-number]-[some-description]`

**Comment by GabrielPadureanu** [comment]
> This would have been marginally nicer if it followed the branch naming convention from the readme: `feat-[repo-name]-[ticket-number]-[some-description]` No point in changing it, but I did say 'better luck next time' last time too if memory serves me well.

Isn't it following the naming convention?
`feat-platform-868-order-and-group-pins`

**Comment by srieding** [comment]
I think I'm blind, looking at the wrong editor window. My bad.

**Review by srieding — APPROVED** [review]

File: src/models/analytics/analytics-tables.ts [hunk 2]
```diff
@@ -401,8 +404,32 @@ const analyticsTablesL10n: Record<TableKeys, AnalyticsTable> = {
     },
     tableKey: 'gridPinGroup',
     tableText: {
-      en: { singular: 'Group-Location', plural: 'Group-Locations' },
-      nl: { singular: 'Groep-Locatie', plural: 'Groep-Locaties' },
+      en: { singular: 'Location-group', plural: 'Location-groups' },
```
Comment (everhardt) [inline src/models/analytics/analytics-tables.ts:407]: I think this was right as it was, it's not a list of grids of type pinGroup, but a list of pinGroups that belong to a grid. That means that a pinGroup can occur multiple times in this table (if it is in multiple grids). Analogous, I think the gridPin table below should be about "group ports", not "port groups".

File: src/routes/graph/add-grid.ts [hunk 3]
```diff
@@ -4,10 +4,12 @@ import { schema as fieldsToServerFullSchema, FieldsToServerFull } from '../../mo
...
 import { schema as gridSchema, Grid } from '../../models/grid';
 
+// TODO srieding deprecated remove after https://github.com/withthegrid/platform-client/issues/1565
+// scope: entire file.
 interface Request {
   body: {
-    fields: FieldsToServerFull;
-    photo?: string;
+  fields: FieldsToServerFull;
+  photo?: string;
```
Comment (everhardt) [inline src/routes/graph/add-grid.ts:12]: There's something wrong with the indentation here (also in the other two add-routes)

File: src/routes/graph/update-pin-grid.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    photo: Joi.string().allow(null).description('Should be a dataurl. Null clears the photo'),
+    pinHashIds: Joi.array().items(Joi.string()).description('Determines the set (and the order) of the pins in the grid'),
+  }).required(),
+  response: Joi.object().keys({
+    grid: gridSchema.required(),
+    pins: Joi.array().items(pinSchema).required(),
+    lastReports: Joi.array().items(Joi.object({
+      pinGroupHashId: Joi.string().required().example('dao97'),
+      generatedAt: Joi.date().required().example('2019-12-31T15:23Z'),
+    })).required(),
```
Comment (everhardt) [inline src/routes/graph/update-pin-grid.ts:45]: lastReports can be deleted I think?

**Review by everhardt — CHANGES_REQUESTED** [review]
Looks good, minor stuff only

File: src/routes/graph/update-pin-grid.ts (hunk 4 above)
Comment (GabrielPadureanu) [reply src/routes/graph/update-pin-grid.ts]: 42713e7d9623803d9754efcd26882dddaba58671

File: src/routes/graph/add-grid.ts (hunk 3 above)
Comment (GabrielPadureanu) [reply src/routes/graph/add-grid.ts:12]: It's fixed

File: src/models/analytics/analytics-tables.ts (hunk 2 above)
Comment (GabrielPadureanu) [reply src/models/analytics/analytics-tables.ts]: Fixed

**Review by everhardt — APPROVED** [review]
👍

**Pull request merged**